### ISSUE TITLE
Error in encoding

### ISSUE THREAD

**Issue by Altai-man** [tony-o/perl6-encoding-huffman-pp6#1]
I have this

```
➜  perl6 --version
This is Rakudo version 2015.12-220-gcf7706f built on MoarVM version 2015.12-29-g8079ca5
implementing Perl 6.c.
```

And error log is

```
> use Encoding::Huffman::PP6;
> huffman-encode("www.example.com");
Use of uninitialized value %huffman-http2 of type Any in string context
Any of .^name, .perl, .gist, or .say can stringify undefined things, if needed.  in block  at /home/sena/.rakudobrew/moar-nom/install/share/perl6/site/sources/22FCCAA3F1C16AA9CF7917097FD218BFEAFD7A1E line 264
Use of uninitialized value %huffman-http2 of type Any in string context
Any of .^name, .perl, .gist, or .say can stringify undefined things, if needed.  in block  at /home/sena/.rakudobrew/moar-nom/install/share/perl6/site/sources/22FCCAA3F1C16AA9CF7917097FD218BFEAFD7A1E line 264
Use of uninitialized value %huffman-http2 of type Any in string context
Any of .^name, .perl, .gist, or .say can stringify undefined things, if needed.  in block  at /home/sena/.rakudobrew/moar-nom/install/share/perl6/site/sources/22FCCAA3F1C16AA9CF7917097FD218BFEAFD7A1E line 264
Use of uninitialized value %huffman-http2 of type Any in string context
Any of .^name, .perl, .gist, or .say can stringify undefined things, if needed.  in block  at /home/sena/.rakudobrew/moar-nom/install/share/perl6/site/sources/22FCCAA3F1C16AA9CF7917097FD218BFEAFD7A1E line 264
Use of uninitialized value %huffman-http2 of type Any in string context
Any of .^name, .perl, .gist, or .say can stringify undefined things, if needed.  in block  at /home/sena/.rakudobrew/moar-nom/install/share/perl6/site/sources/22FCCAA3F1C16AA9CF7917097FD218BFEAFD7A1E line 264
Use of uninitialized value %huffman-http2 of type Any in string context
Any of .^name, .perl, .gist, or .say can stringify undefined things, if needed.  in block  at /home/sena/.rakudobrew/moar-nom/install/share/perl6/site/sources/22FCCAA3F1C16AA9CF7917097FD218BFEAFD7A1E line 264
Use of uninitialized value %huffman-http2 of type Any in string context
Any of .^name, .perl, .gist, or .say can stringify undefined things, if needed.  in block  at /home/sena/.rakudobrew/moar-nom/install/share/perl6/site/sources/22FCCAA3F1C16AA9CF7917097FD218BFEAFD7A1E line 264
Use of uninitialized value %huffman-http2 of type Any in string context
Any of .^name, .perl, .gist, or .say can stringify undefined things, if needed.  in block  at /home/sena/.rakudobrew/moar-nom/install/share/perl6/site/sources/22FCCAA3F1C16AA9CF7917097FD218BFEAFD7A1E line 264
Use of uninitialized value %huffman-http2 of type Any in string context
Any of .^name, .perl, .gist, or .say can stringify undefined things, if needed.  in block  at /home/sena/.rakudobrew/moar-nom/install/share/perl6/site/sources/22FCCAA3F1C16AA9CF7917097FD218BFEAFD7A1E line 264
Use of uninitialized value %huffman-http2 of type Any in string context
Any of .^name, .perl, .gist, or .say can stringify undefined things, if needed.  in block  at /home/sena/.rakudobrew/moar-nom/install/share/perl6/site/sources/22FCCAA3F1C16AA9CF7917097FD218BFEAFD7A1E line 264
Use of uninitialized value %huffman-http2 of type Any in string context
Any of .^name, .perl, .gist, or .say can stringify undefined things, if needed.  in block  at /home/sena/.rakudobrew/moar-nom/install/share/perl6/site/sources/22FCCAA3F1C16AA9CF7917097FD218BFEAFD7A1E line 264
Use of uninitialized value %huffman-http2 of type Any in string context
Any of .^name, .perl, .gist, or .say can stringify undefined things, if needed.  in block  at /home/sena/.rakudobrew/moar-nom/install/share/perl6/site/sources/22FCCAA3F1C16AA9CF7917097FD218BFEAFD7A1E line 264
Use of uninitialized value %huffman-http2 of type Any in string context
Any of .^name, .perl, .gist, or .say can stringify undefined things, if needed.  in block  at /home/sena/.rakudobrew/moar-nom/install/share/perl6/site/sources/22FCCAA3F1C16AA9CF7917097FD218BFEAFD7A1E line 264
Use of uninitialized value %huffman-http2 of type Any in string context
Any of .^name, .perl, .gist, or .say can stringify undefined things, if needed.  in block  at /home/sena/.rakudobrew/moar-nom/install/share/perl6/site/sources/22FCCAA3F1C16AA9CF7917097FD218BFEAFD7A1E line 264
Use of uninitialized value %huffman-http2 of type Any in string context
Any of .^name, .perl, .gist, or .say can stringify undefined things, if needed.  in block  at /home/sena/.rakudobrew/moar-nom/install/share/perl6/site/sources/22FCCAA3F1C16AA9CF7917097FD218BFEAFD7A1E line 264
Buf[uint8]:0x<ff ff ff fc>
```

It seems encoding was broken due to some internal changes in rakudo. I'm writing a hpack compression module currently and it will be good to fix this issue.

**Comment by Altai-man** [comment]
When I redefined the table with values like `0.chr => "1111111111000",` encoding was successful. But I don't know what to do with `_eos`. Anyway, I will add PR with this quick fix later.

**Comment by Altai-man** [comment]
Can be closed due to #2.
